base/loading.jl

```diff
@@ -1677,6 +1677,7 @@ function _include_dependency(mod::Module, _path::AbstractString; track_content=t
                 push!(_require_dependencies,
                       (mod, path, filesize(path), open(_crc32c, path, "r"), -1.0))
             else
+                @assert ispath(path) "'$path': No such file or directory"
```

Comment:
```suggestion
                (isfile(path) && uperm(path) & 0x04 == 0x04) ||
                    throw(ArgumentError("$(repr(path)) is not readable"))
```

Reply:
The perm check should be moved to the other branch, because `incluce_dependency`s are only tracked by `mtime`, or does that also require read permissions?

Reply:
#51798 is concerned with the other branch. If the file does not exist there, then the error is thrown through `open()`.